### PR TITLE
Fix: Improving memory management when loading scenes

### PR DESCRIPTION
**Pull request by 0xD-Fabio**
## What does this PR change?

This PR enforces some objects' references to null to limit memory garbage accumulated while doing multiple teleporting from one scene to another.

...

## How to test the changes?

0. Go to https://play.decentraland.zone/?renderer-branch=fix/scene-loading-leak
1. Try some sequential teleport from Genesis Plaza to other locations and back, waiting a bit between teleports
2. Check for crashes if the platform is stable even between multiple teleports
3. hopefully, enjoy a bit less filled RAM 

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by kwsskulinski — APPROVED** [review]
The issue appears to be fixed. There were no crashes and RAM usage manageable.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Textures/DCLTexture.cs [hunk 1]
```diff
@@ -206,7 +206,7 @@ public override void Dispose()
                 RemoveReference(attachedEntitiesByComponent.First().Key);
             }
 
-            if (texturePromise != null)
+            if (texturePromise != null && AssetPromiseKeeper_Texture.i != null)
```
Comment (AjimenezDCL) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Textures/DCLTexture.cs:209]: `AssetPromiseKeeper_Texture.i` should never be null, it's auto instantiated when requested.

**Review by AjimenezDCL — CHANGES_REQUESTED** [review]

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Textures/DCLTexture.cs (hunk 1 above)
Comment (0xD-Fabio) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Textures/DCLTexture.cs]: Thanks, it seemed to make a test fail at some point, but it wasn't.

**Review by AjimenezDCL — APPROVED** [review]
Good job!!

**Pull request merged**